### PR TITLE
Read editor overlays from XML

### PR DESCRIPTION
**Pull request by Kert**

### CODE REVIEW

File: src/cpp/shared/adventure/map.h [hunk 1]
```diff
@@ -325,6 +325,7 @@ enum ADVENTURE_MAP_LOCATION {
   LOCATION_RANDOM_MONSTER_VERY_STRONG = 54,
   LOCATION_RANDOM_HERO = 55,
   LOCATION_NOTHING_SPECIAL = 56,
+  LOCATION_WHAT_IS_THIS = 57,
```
Comment (jkoppel) [inline src/cpp/shared/adventure/map.h]: Is this used?

**Comment by jkoppel** [comment]
Please confirm that the overlays array obtained from doing this is binary-identical to the original one.

Once you have done so, you may merge.

File: src/cpp/shared/adventure/map.h (hunk 1 above)
Comment (Kert) [reply src/cpp/shared/adventure/map.h]: It is used in overlays array

**Comment by jkoppel** [comment]
What overlay uses LOCATION_WHAT_IS_THIS?

**Comment by Kert** [comment]
lol. There is a literally a screenshot of objects using this type above in this PR

**Comment by jkoppel** [comment]
Oh oh oh oh oh oh.

**Comment by jkoppel** [comment]
How about LOCATION_HOLE_LIKE ?

**Comment by Kert** [comment]
> Please confirm that the overlays array obtained from doing this is binary-identical to the original one.
> 
> Once you have done so, you may merge.

Checked. Original array and the one read from XML are binary-identical

**Pull request merged**